File: hotels-backend/Hotels/DTOs/ReservationDTO.cs
using Hotels.JsonConverters;
using Hotels.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Hotels.DTOs
{
    public partial class ReservationDTO
    {
        public int Id { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        [JsonConverter(typeof(TimeSpanConverter))]
        public TimeSpan ArrivalTime { get; set; }

        [JsonConverter(typeof(TimeSpanConverter))]
        public TimeSpan DepartureTime { get; set; }

        public string HotelName { get; set; }
        public string Country { get; set; }
        public string City { get; set; }
        public string RoomNumber { get; set; }

        public int RoomId { get; set; }

        public virtual ICollection<ReservationService> ReservationServices { get; set; }
    }
}

File: hotels-admin-frontend/src/app/reservation-dto.ts
import { ReservationService } from "./reservationService";

export class ReservationDTO {
  id: number;
  roomId: number;
  userId: number;
  startDate: Date;
  endDate: Date;
  arrivalTime: string;
  departureTime: string;
  country: string;
  city: string;
  cost: number;
  hotelName: string;
  roomNumber: string;
  userName: string;
  reservationServices: ReservationService[];


  constructor() {
      this.id = 0;
      this.roomId = 0;
      this.userId = 0;
      this.startDate = new Date();
      this.endDate = new Date();
      this.arrivalTime = '00:00';
      this.departureTime = '00:00';
      this.country = '';
      this.city = '';
      this.cost = 0;
      this.hotelName = '';
      this.roomNumber = '';
      this.userName = '';
      this.reservationServices = [];
  }
}